### DISCUSSION TITLE
Sqlite如何使用InsertOrUpdate类似于指定临时主键功能

### DISCUSSION THREAD

**Discussion by Thinking-319** [dotnetcore/FreeSql#1669] [Q&A] [answered]
在Sqlite中使用`InsertOrUpdate`时，指定某一临时主键`Name`，`dataList`中有多条数据，在实际插入的过程数据库只会新增一条记录，并不断更新数据。
```C#
public static int NewInsertDataToDb(List<TestData> dataList)
{
     if (dataList == null || dataList.Count == 0)  return -1;
     return Db.SqlServer.InsertOrUpdate<TestData>().SetSource(dataList, a => a.Name).IfExistsDoNothing().ExecuteAffrows(); 
}
```
根据pr（[pr链接](https://github.com/dotnetcore/FreeSql/pull/1160)）中的提示《**不处理自增，因某些数据库依赖主键或唯一键，所以指定临时主键仅对 SqlServer/PostgreSQL/Firebird/达梦/南大通用/金仓/神通 有效》**

**如果在Sqlite中想使用相同的功能，该如何实现呢？**
即：当待插入数据的某一字段或组合字段与数据库中已有记录相同时，不进行操作或者更新当前数据，当不相同时则插入数据。

![image](https://github.com/dotnetcore/FreeSql/assets/94663118/23bcd2ac-0663-4944-a3aa-00f8d61d095b)

**Comment by 2881099** [comment, marked as answer]
使用前先把id查询出来，若有则填充到实体上，再进行操作
